src/lib/virtualnode/message_handler.ts

```diff
@@ -0,0 +1,35 @@
+import { Driver } from "zwave-js";
+import { MissingInputParameterError, UnknownCommandError } from "../error";
+import { VirtualNodeCommand } from "./command";
+import { IncomingMessageVirtualNode } from "./incoming_message";
+import { VirtualNodeResultTypes } from "./outgoing_message";
+
+export class VirtualNodeMessageHandler {
+  static async handle(
+    message: IncomingMessageVirtualNode,
+    driver: Driver
+  ): Promise<VirtualNodeResultTypes[VirtualNodeCommand]> {
+    const { command } = message;
+
+    if (!message.broadcast && !message.nodeIDs) {
+      throw new MissingInputParameterError(command, ["broadcast", "nodeIDs"]);
+    }
+    const virtualNode = message.nodeIDs
+      ? driver.controller.getMulticastGroup(message.nodeIDs)
+      : driver.controller.getBroadcastNode();
```

Comment:
I don't think that this makes sense. We should try to map our APIs 1-1 with ZJS. So this should be 2 controller commands instead.

Reply:
See latest commit - is this what you mean?

Reply:
Just reread your comment. The problem with making them controller commands is that they either have to be stored on the server for future access or they have to be recreated before calling any of the methods, they aren't persisted anywhere in the driver or controller

Reply:
ZJS Server should not offer any extra logic/persistence on top of ZJS. It should just be a proxy. Let clients handle with it.

Reply:
I don't understand how we can let the client deal with a virtual node. There's nothing we can serialize to pass the node back and forth between client and server